### PR TITLE
Add systemd_exporter

### PR DESCRIPTION
**Pull request by JGodin-C2C**
Add Systemd_exporter

# What:

Simply add the possbility to install the systemd_exporter hosted here : 

https://github.com/prometheus-community/systemd_exporter

Pull request have been tested.

### CODE REVIEW

File: data/defaults.yaml [hunk 1]
```diff
@@ -236,6 +236,13 @@ prometheus::statsd_exporter::package_name: 'statsd_exporter'
 prometheus::statsd_exporter::service_name: 'statsd_exporter'
 prometheus::statsd_exporter::user: 'statsd-exporter'
 prometheus::statsd_exporter::version: '0.17.0'
+prometheus::systemd_exporter::download_extension: 'tar.gz'
```
Comment (bastelfreak) [inline data/defaults.yaml]: please add the defaults in the class itself, not in hiera. That's an old pattern we don't want to use anymore. adding it in the class enables puppet-strings to pick up the data and put it into the REFERENCE.md

File: manifests/alertmanager.pp [hunk 2]
```diff
@@ -157,10 +157,10 @@
   }
 
   $alertmanager_reload = $prometheus::init_style ? {
-    'systemd'          => "systemctl reload-or-restart ${service_name}",
-    /^(upstart|none)$/ => "service ${service_name} reload",
-    /^(sysv|sles)$/    => "/etc/init.d/${service_name} reload",
-    'launchd'          => "launchctl stop ${service_name} && launchctl start ${service_name}",
+    'systemd'              => "systemctl reload-or-restart ${service_name}",
+    /^(upstart|none)$/     => "service ${service_name} reload",
+    /^(sysv|redhat|sles)$/ => "/etc/init.d/${service_name} reload",
```
Comment (bastelfreak) [inline manifests/alertmanager.pp]: that doesn't look correct to me, shouldn't they use systemd?

File: data/defaults.yaml (hunk 1 above)
Comment (JGodin-C2C) [reply data/defaults.yaml]: Noted, the code have been changed accordingly. :+1:

File: manifests/alertmanager.pp (hunk 2 above)
Comment (JGodin-C2C) [reply manifests/alertmanager.pp]: This was part of a 'too violent rebase'.

**Comment by JGodin-C2C** [comment]
Eeeer, If i am not mistaking, build errors seems to be part of another portion of the code ?

File: manifests/alertmanager.pp [hunk 3]
```diff
@@ -59,6 +59,8 @@
 #  If package, then use this for package ensure default 'latest'
 # @param package_name
 #  The binary package name - not available yet
+# @param config_dir
```
Comment (bastelfreak) [inline manifests/alertmanager.pp]: can you please put the not systemd_exporter changes into a dedicated PR?

Comment (JGodin-C2C) [reply manifests/alertmanager.pp]: I can do that !
